go/vt/vtgate/engine/merge_sort_test.go

```diff
@@ -342,7 +342,7 @@ func (t *fakeVcursor) Execute(method string, query string, bindvars map[string]*
 	panic("unimplemented")
 }
 
-func (t *fakeVcursor) ExecuteMultiShard(keyspace string, shardQueries map[string]*querypb.BoundQuery, isDML bool) (*sqltypes.Result, error) {
+func (t *fakeVcursor) ExecuteMultiShard(keyspace string, shardQueries map[string]*querypb.BoundQuery, isDML, isFinal bool) (*sqltypes.Result, error) {
```

Comment:
I feel like `isFinal` is non-obvious -- it seems better to use something more explicit like `autocommitSafe` which would capture things more clearly.

Reply:
This comment was held in the GH Review but I mistakenly sent some as single comments. Nevertheless it seems like you agree and I think `canCommit` is a sensible name.